src/test/suite/completion.test.ts

```diff
@@ -392,6 +393,22 @@ describe("Should do completion", () => {
       )
     ).twice();
   });
+  it("should report suggestion about to shown after debounce", async () => {
```

Comment:
please add a test that the event is not fired if the completion has been canceled (i.e only send events on completions that are actually about to be shown and not canceled)